### PR TITLE
CRS reports: Functionality checklist report

### PR DESCRIPTION
**Pull request by agrzywinski**
@snopoke

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: custom/up_nrhm/sql_data.py [hunk 1]
```diff
@@ -198,3 +198,89 @@ def filters(self):
...
+    def table_name(self):
+        return get_table_name(self.config['domain'], TABLE_ID)
+
+    def _qualify_column(self, column):
+        return '"{}".{}'.format(self.table_name, column)
+
+    @property
+    def columns(self):
+        return [
+            DatabaseColumn("Total number of ASHAs under the Facilitator", SimpleColumn("case_id",)),
```
Comment (snopoke) [inline custom/up_nrhm/sql_data.py]: There is a `doc_id` column which you could use which wouldn't require any grouping and could then also serve as the filter in `ASHAAFChecklistData`.

File: custom/up_nrhm/sql_data.py [hunk 2]
```diff
@@ -198,3 +198,89 @@ def filters(self):
...
+
+
+class ASHAFunctionalityChecklistData(SqlData):
+    slug = 'asha_functionality_checklist'
+
+    @property
+    def table_name(self):
+        return get_table_name(self.config['domain'], TABLE_ID)
+
+    def _qualify_column(self, column):
```
Comment (snopoke) [inline custom/up_nrhm/sql_data.py]: Doesn't seem like this gets used, or `table_name`

Comment (snopoke) [inline custom/up_nrhm/sql_data.py]: Same in `ASHAAFChecklistData`

File: custom/up_nrhm/reports/asha_facilitators_report.py [hunk 3]
```diff
@@ -9,8 +9,8 @@
 
 def total_rows(report):
     return {
-        "total_under_facilitator": report.total_under_facilitator,
-        "total_with_checklist": report.total_with_checklist
+        "total_under_facilitator": getattr(report, 'total_under_facilitator', 0),
+        "total_with_checklist": getattr(report, 'total_with_checklist', 0)
```
Comment (snopoke) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: Was this causing errors?

**Comment by snopoke** [comment]
Tested locally and looks OK so I'm going to merge but please review my comments.

File: custom/up_nrhm/reports/asha_facilitators_report.py (hunk 3 above)
Comment (agrzywinski) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: @snopoke I had a attribute error when I opened a report and filters were not selected
I'll fix things that you mentioned in other comments in pr with next report.

Comment (snopoke) [inline custom/up_nrhm/reports/asha_facilitators_report.py]: :+1:

**Pull request merged**